lib/ansible/modules/packaging/os/apt.py

```diff
@@ -535,6 +554,24 @@ def install(m, pkgspec, cache, upgrade=False, default_release=None,
     packages = ' '.join(pkg_list)
 
     if packages:
+        policy_rc_file = '/usr/sbin/policy-rc.d'
+
+        if autostart is False and not os.path.isfile(policy_rc_file):
+            e = None
+
+            try:
+                f = open(policy_rc_file, 'w')
+                f.write("#!/bin/bash\nexit 101\n")
+                os.chmod(policy_rc_file, 0o755)
+                f.close()
+            except:
+                return False, dict(msg="Cannot create file %s: %s" % (policy_rc_file, e))
+        elif autostart is True and os.path.isfile(policy_rc_file):
+            try:
+                os.remove(policy_rc_file)
```

Comment:
This is not concurrency safe.

Reply:
It seems the [systemd mask](https://major.io/2016/05/05/preventing-ubuntu-16-04-starting-daemons-package-installed/) is a better solution since the `RUNLEVEL=1` doesn't work.

Reply:
@tonyseek this needs to work no matter what the init system is.

it should be concurrency 'safe' since the dpkg db requires exclusive locking